### PR TITLE
Some work to remove JDBI to make a native Quarkus build possible

### PR DESCRIPTION
**Pull request by EricWittmann**
The strategy here was to replicate the same patterns used by JDBI, but with custom interfaces/classes.  As I suspected, the surface area wasn't really that large.

I also found some interesting improvements, especially using `findOne()` and checking for empty (to detect 404 errors) rather than using `find()` and catching an `IllegalStateException`.  Give this a look and let me know what you think!

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java [hunk 1]
```diff
@@ -1644,18 +1653,23 @@ public ArtifactVersionMetaDataDto getArtifactVersionMetaData(String groupId, Str
...
-                return handle.createQuery(sql)
+                Optional<ArtifactVersionMetaDataDto> res = handle.createQuery(sql)
                         .bind(0, tenantContext.tenantId())
                         .bind(1, normalizeGroupId(groupId))
                         .bind(2, artifactId)
                         .bind(3, version)
                         .map(ArtifactVersionMetaDataDtoMapper.instance)
-                        .one();
+                        .findOne();
+                if (res.isEmpty()) {
```
Comment (carlesarnal) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java:1665]: Just a preference of mine, but I usually prefer to check the positive case, with `ifPresent`. I find it more readable.

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java [hunk 2]
```diff
@@ -1518,18 +1522,23 @@ public StoredArtifactDto getArtifactVersion(String groupId, String artifactId, S
...
+                Optional<StoredArtifactDto> res = handle.createQuery(sql)
                         .bind(0, tenantContext.tenantId())
                         .bind(1, normalizeGroupId(groupId))
                         .bind(2, artifactId)
                         .bind(3, version)
                         .map(StoredArtifactMapper.instance)
-                        .one();
+                        .findOne();
+                if (res.isEmpty()) {
+                    throw new ArtifactNotFoundException(groupId, artifactId);
```
Comment (carlesarnal) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java:1535]: Same here with the `Optional` suggestion.

**Comment by carlesarnal** [comment]
This looks like a good start to me. Just that minor suggestion for Optionals to simplify the code a bit. There are a few more places where that happens than the ones where I have commented.

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java (hunk 1 above)
Comment (EricWittmann) [reply app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java]: Very good suggestion using `orElseThrow`.  I have changed all the places where this was relevant.

**Comment by EricWittmann** [comment]
Interestingly, the build is having some **compilation** failures which I can't understand.  The build works locally for me on both of my machines.  The compile failures seem to point to irrelevant lines of code.  I've pushed a new commit to the branch to see if that makes a difference...

**Pull request closed without merging**